metricq/agent.py

```diff
@@ -142,21 +142,10 @@ def derive_address(self, address: str) -> str:
 
     @property
     def event_loop(self) -> asyncio.AbstractEventLoop:
-        # TODO maybe we should instead **TRY** to get the loop of the current context
         if self._event_loop is None:
-            logger.debug("Getting current or new")
-            try:
-                self._event_loop = asyncio.get_event_loop()
-            except RuntimeError:
-                # Until python 3.6, it may raise instead of creating a new one
-                self._event_loop = asyncio.new_event_loop()
+            self._event_loop = asyncio.get_event_loop()
```

Comment:
```
def _event_loop(self) -> asyncio.AbstractEventLoop:
    loop = asyncio.get_running_loop()
    if self._loop is None:
        with _global_thread_lock:
            if self._loop is None:
               self._loop = loop
    if loop is not self._loop:
        raise RuntimeError(f'{self!r} is bound to a different event loop')
    return loop
```

+ a module level `_global_thread_lock = threading.Lock()` (that's how asyncio does it :shrug:)
+ `_loop = None` initialization.

And then *always* use this when we need a loop internally.